### PR TITLE
code: add unit test for restorer

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Add unit test for restorer

### Check List 

Tests 

 - Unit test

Side effects

N/A

Related changes

N/A

### CODE REVIEW

**Comment by lonng** [comment]
PTAL @amyangfei @kennytm

File: code/rewriter_test.go [hunk 1]
```diff
@@ -1328,22 +1340,31 @@ func unittest() {
...
-		c.Assert(err, Equals, nil)
+		c.Assert(err, IsNil)
 	}()
 
 	rewriter := code.NewRewriter(s.path)
 	err = rewriter.Rewrite()
+	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Matches, "failpoint.Inject: invalid signature with type.*")
+
+	// Workspace should be keep clean if some error occurred
```
Comment (amyangfei) [inline code/rewriter_test.go]:
```suggestion
	// Workspace should keep clean if some error occurred
```

File: code/rewriter_test.go [hunk 2]
```diff
@@ -1374,23 +1395,31 @@ func unittest() {
...
-		c.Assert(err, Equals, nil)
+		c.Assert(err, IsNil)
 	}()
 
 	rewriter := code.NewRewriter(s.path)
 	err = rewriter.Rewrite()
 	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Matches, "failpoint.Inject: invalid signature.*")
+
+	// Workspace should be keep clean if some error occurred
```
Comment (amyangfei) [inline code/rewriter_test.go]:
```suggestion
	// Workspace should keep clean if some error occurred
```

File: code/rewriter_test.go [hunk 3]
```diff
@@ -1421,21 +1450,29 @@ func unittest() {
...
```
Comment (amyangfei) [inline code/rewriter_test.go]:
```suggestion
	// Workspace should keep clean if some error occurred
```

File: code/rewriter_test.go [hunk 4]
```diff
@@ -1328,22 +1340,31 @@ func unittest() {
...
 	err = rewriter.Rewrite()
+	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Matches, "failpoint.Inject: invalid signature with type.*")
+
+	// Workspace should be keep clean if some error occurred
+	for _, cs := range cases {
+		original := filepath.Join(s.path, cs.filepath)
+		content, err := ioutil.ReadFile(original)
+		c.Assert(err, IsNil)
+		c.Assert(strings.TrimSpace(string(content)), Equals, strings.TrimSpace(cs.original))
```
Comment (amyangfei) [inline code/rewriter_test.go]: `strings.TrimSpace` seems not necessary in all the following `restore test`, as `content` is restored from stash file, it should and must have the same content before rewriting.

Comment (lonng) [reply code/rewriter_test.go]: addressed

File: code/rewriter_test.go [hunk 5]
```diff
@@ -1328,22 +1340,31 @@ func unittest() {
...
 	defer func() {
 		err := os.RemoveAll(s.path)
-		c.Assert(err, Equals, nil)
+		c.Assert(err, IsNil)
 	}()
 
 	rewriter := code.NewRewriter(s.path)
 	err = rewriter.Rewrite()
+	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Matches, "failpoint.Inject: invalid signature with type.*")
```
Comment (kennytm) [inline code/rewriter_test.go]:
```suggestion
	c.Assert(err, ErrorMatches, `failpoint\.Inject: invalid signature with type.*`)
```

Also remove the `c.Assert(err, NotNil)` above

File: .codecov.yml [hunk 6]
```diff
@@ -31,7 +31,9 @@ comment:
 
 ignore:
   - "LICENSES"
-  - "*_test.go"       
+  - "*_test.go"
+  - "marker.go"
+  - "failpoint-ctl"
```
Comment (kennytm) [inline .codecov.yml]: Add a comment to explain why these two entries are added.

File: .codecov.yml [hunk 7]
```diff
@@ -31,7 +31,9 @@ comment:
 
 ignore:
   - "LICENSES"
-  - "*_test.go"       
+  - "*_test.go"
+  - "marker.go"
```
Comment (lonng) [inline .codecov.yml]:
```suggestion
  - "marker.go" # This file only contains empty function stub
```

Comment (kennytm) [reply .codecov.yml]: 👍

File: .codecov.yml (hunk 6 above)
Comment (lonng) [inline .codecov.yml]:
```suggestion
  - "failpoint-ctl" # Ignore the `failpoint-ctl` command line tool
```

Comment (kennytm) [reply .codecov.yml]: But how come this is included in the first place 🤔

**Pull request merged**